TST: DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays

Started seeing this failure in dev job:

```
Numpy: 1.24.0.dev0+934.gdb7414b7f
...
____________________________ TestCore.test_byteswap ____________________________

self = <astropy.io.fits.tests.test_core.TestCore object at 0x7f4c26839690>

    def test_byteswap(self):
        p = fits.PrimaryHDU()
        lst = fits.HDUList()
    
        n = np.zeros(3, dtype='i2')
        n[0] = 1
>       n[1] = 60000
E       DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays.
        The conversion of 60000 to int16 will fail in the future.
E       For the old behavior, usually:
E           np.array(value).astype(dtype)`
E       will give the desired result (the cast overflows).

.../astropy/io/fits/tests/test_core.py:56: DeprecationWarning
```

This affected line was copied over from PyFITS and has not been touched for 11 years. Why now, NumPy? 😿 

https://github.com/astropy/astropy/blob/06e66c303ed21114d7b39584ca76dc09c60902ca/astropy/io/fits/tests/test_core.py#L56

--- a/astropy/io/fits/tests/test_core.py
+++ b/astropy/io/fits/tests/test_core.py
@@ -51,7 +51,7 @@ class TestCore(FitsTestCase):
         p = fits.PrimaryHDU()
         lst = fits.HDUList()
 
-        n = np.array([1, 60000, 0]).astype('i2')
+        n = np.array([1, 60000, 0], dtype='u2').astype('i2')
 
         c = fits.Column(name='foo', format='i2', bscale=1, bzero=32768,
                         array=n)

--- a/astropy/io/fits/tests/test_core.py
+++ b/astropy/io/fits/tests/test_core.py
@@ -51,10 +51,7 @@ class TestCore(FitsTestCase):
         p = fits.PrimaryHDU()
         lst = fits.HDUList()
 
-        n = np.zeros(3, dtype='i2')
-        n[0] = 1
-        n[1] = 60000
-        n[2] = 2
+        n = np.array([1, 60000, 0]).astype('i2')
 
         c = fits.Column(name='foo', format='i2', bscale=1, bzero=32768,
                         array=n)

